crates/ruff_linter/src/rules/flake8_use_pathlib/rules/dotless_pathlib_with_suffix.rs

```diff
@@ -46,14 +47,16 @@ use ruff_text_size::Ranged;
 #[derive(ViolationMetadata)]
 pub(crate) struct DotlessPathlibWithSuffix;
 
-impl AlwaysFixableViolation for DotlessPathlibWithSuffix {
+impl Violation for DotlessPathlibWithSuffix {
+    const FIX_AVAILABILITY: FixAvailability = FixAvailability::Sometimes;
+
     #[derive_message_formats]
     fn message(&self) -> String {
         "Dotless suffix passed to `.with_suffix()`".to_string()
     }
 
-    fn fix_title(&self) -> String {
-        "Add a leading dot".to_string()
+    fn fix_title(&self) -> Option<String> {
+        Some("Add a leading dot".to_string())
```

Comment:
We need to update the fix title for the `.` case because the fix is always shown as help text and *add a leading dot* is misleading for `.`